examples/grpc/src/main.rs

```diff
@@ -0,0 +1,162 @@
+use hyper::server::accept::from_stream;
+use hyper::service::make_service_fn;
+use hyper::{Client, Server, Uri};
+use std::convert::Infallible;
+use std::net::{IpAddr, Ipv4Addr};
+use tonic::Status;
+use tonic::{Request, Response};
+use tracing::{info_span, Instrument, Span};
+use turmoil::{net, Builder};
+
+mod proto {
+    tonic::include_proto!("helloworld");
+}
+
+use proto::greeter_server::{Greeter, GreeterServer};
+use proto::{HelloReply, HelloRequest};
+
+use crate::proto::greeter_client::GreeterClient;
+
+fn main() {
+    if std::env::var("RUST_LOG").is_err() {
+        std::env::set_var("RUST_LOG", "info");
+    }
+
+    tracing_subscriber::fmt::init();
+
+    let addr = (IpAddr::from(Ipv4Addr::UNSPECIFIED), 9999);
+
+    let mut sim = Builder::new().build();
+
+    let greeter = GreeterServer::new(MyGreeter {});
+
+    sim.host("server", move || {
+        let greeter = greeter.clone();
+        async move {
+            let listener = net::TcpListener::bind(addr).await.unwrap();
```

Comment:
You can avoid the unwraps in the server now, hosts now also fail the sim if they err.